tests/CommandBusTest.php

```diff
@@ -67,4 +68,24 @@ public function testHandleThrowExceptionForInvalidCommand()
     {
         (new CommandBus([]))->handle('must be an object');
     }
+
+    public function testIfOneCanOnlyCreateWithValidMiddlewares()
+    {
+        $middlewareList = [$this->getMock('stdClass')];
+
+        try {
+            new CommandBus($middlewareList);
+            $this->fail('Should not be able to create a CommandBus without a valid list of middlewares!');
+        } catch (InvalidMiddlewareException $caught) {
```

Comment:
It is still possible since the exception check is done once the is finished.
```PHP
$this->setExpectedException('League\Tactician\Exception\InvalidMiddlewareException');

$middlewareList = [$this->getMock('stdClass')];
new CommandBus($middlewareList);
$this->fail('Should not be able to create a CommandBus without a valid list of middlewares!');
```